erpnext/patches/v15_0/update_warehouse_field_in_asset_repair_consumed_item_doctype.py

```diff
@@ -0,0 +1,12 @@
+import frappe
+
+
+def execute():
+	if frappe.db.has_column("Asset Repair", "warehouse"):
+		items = frappe.get_all("Asset Repair Consumed Item", fields=["name", "parent"])
+
+		for item in items:
+			warehouse = frappe.db.get_value("Asset Repair", item.parent, "warehouse")
+			if warehouse:
+				frappe.db.set_value("Asset Repair Consumed Item", item.name, "warehouse", warehouse)
```

Comment:
We can avoid the `for` loop and multiple update queries, by using a simple join query (between parent and children) to update the value.

```
frappe.db.sql("""
	update `tabAsset Repair` ar, `Asset Repair Consumed Item` ar_item
	set ar_item.warehouse = ar.warehouse
	where ar.name = ar_item.parent
		and ifnull(ar.warehouse, '') != ''
""")
```

Use `frappe.qb` instead of `frappe.db.sql`